src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -402,10 +402,14 @@ def link_content(text, css_class, icon)
     end
   end
 
-  def user_notification_link(link_text, new_filter, filter: 'inbox')
+  def user_notification_link(link_text, new_filter)
     css_class = 'list-group-item list-group-item-action'
-    css_class += ' active' if new_filter == filter || (filter.nil? && new_filter == 'inbox')
-    link_to(link_text, my_notifications_path(type: new_filter), class: css_class)
+    if (new_filter[:type].present? && params[:project].nil? && new_filter[:type] == params[:type]) ||
+       (new_filter[:project].present? && new_filter[:project] == params[:project]) ||
+       (params[:type].nil? && params[:project].nil? && new_filter[:type] == 'inbox')
```

Comment:
any way to make this easier to parse for humans?

Reply:
Done. Please have a look again.